packages/components/src/DataList/components/DataListHeaderTile/DataListHeaderTile.tsx

```diff
@@ -30,9 +30,11 @@ export function DataListHeaderTile<T extends DataListObject>({
       onClick={handleOnClick}
     >
       <Text maxLines="single">{headers[headerKey]}</Text>
-      {sortingState?.key === headerKey && (
+      {sortingState?.key === headerKey ? (
         <DataListSortingArrows order={sortingState.order} />
-      )}
+      ) : sortingState?.key !== headerKey && isSortable ? (
```

Comment:
This **feels** like it could be neater, but I also understand what's happening just fine so I'm going to allow it. :)